KakapoExample/KakapoTests/SerializerTests.swift

```diff
@@ -0,0 +1,155 @@
+                checkObject(friends)
+            }
+            
+            it("serialize dictionary and entities inside it") {
+                let friend = Friend(friends: [user])
+                let serialized = serialize(friend)
+                let dictionary = serialized["dictionary"] as? [String: Any]
+                let friends = dictionary?["test"] as? [Any]
+                checkObject(friends)
+            }
+        }
+        
+        describe("Property policy serialization") { 
+            it("is not serialized if nil") {
+                let empty = MaybeEmpty(IgnorableNilProperty<Int>(nil))
+                let serialized = serialize(empty)
+                expect(serialized.count).to(be(0))
+            }
+            
+            it("is serialized if not nil") {
+                let notEmpty = MaybeEmpty(IgnorableNilProperty(1))
+                let serialized = serialize(notEmpty)
+                let value = serialized["value"] as? Int
+                expect(value).to(be(1))
+            }
+            
+            it("recursively serialize the object if needed") {
+                let notEmpty = MaybeEmpty(IgnorableNilProperty(user))
+                let serialized = serialize(notEmpty)
+                let value = serialized["value"] as? [String: Any]
+                expect(value?["name"] as? String).to(equal("Alex"))
+            }
+
+            it("recursively serialize IgnorableNilProperties") {
+                let notEmpty = MaybeEmpty(IgnorableNilProperty(IgnorableNilProperty(1)))
+                let serialized = serialize(notEmpty)
+                let value = serialized["value"] as? Int
+                expect(value).to(be(1))
+            }
+        }
+        
+        describe("Optional property serialization") { 
+            it("serialize nil") {
+                let nilInt: Int? = nil
+                let optional = MaybeEmpty(nilInt)
+                let serialized = serialize(optional)
+                expect(serialized["value"] as? NSNull).to(be(NSNull()))
+            }
+            
+            it("serialize an optional") {
+                let optional = MaybeEmpty(Optional.Some(1))
+                let serialized = serialize(optional) as? [String: Int]
+                expect(serialized?["value"]).to(be(1))
+            }
+            
+            it("recursively serialize the value") {
+                let optional = MaybeEmpty(Optional.Some(user))
+                let serialized = serialize(optional)
+                let value = serialized["value"] as? [String: Any]
+                expect(value?["name"] as? String).to(equal("Alex"))
+            }
+            
+            it("recursively serialize Optionals") {
+                let optional = MaybeEmpty(Optional.Some(Optional.Some(1)))
+                let serialized = serialize(optional) as? [String: Int]
+                expect(serialized?["value"]).to(be(1))
+            }
+        }
+    }
```

Comment:
The tests for optionals and IgnorableNilProperty are really good, though don't you think we need deeper tests for arrays and dictionaries? I am missing some tests with numerous items, both for arrays and dictionaries.